flink/v1.19/flink/src/test/java/org/apache/iceberg/flink/sink/committer/TestIcebergCommitter.java

```diff
@@ -0,0 +1,1452 @@
+
+  private static class ForwardingCommitter extends IcebergCommitter {
+    private int successfulCommits = 0;
+
+    ForwardingCommitter(TableLoader tableLoader) {
+      super(tableLoader, null, Maps.newHashMap(), false, 2, "test-sink-prefix", null);
+    }
+
+    @Override
+    public void commit(Collection<CommitRequest<IcebergCommittable>> commitRequests)
+        throws IOException, InterruptedException {
+      commitRequests.forEach(
+          one -> LOG.info("Committed with numberOfRetries: {}", one.getNumberOfRetries()));
+      successfulCommits += commitRequests.size();
+    }
+
+    @Override
+    public void close() {}
+
+    public int getSuccessfulCommits() {
+      return successfulCommits;
+    }
+  }
+
+  // ------------------------------- Utility Methods --------------------------------
+
+  private IcebergCommitter getCommitter() {
+    IcebergFilesCommitterMetrics metric = mock(IcebergFilesCommitterMetrics.class);
+    return new IcebergCommitter(
+        tableLoader,
+        branch,
+        Collections.singletonMap("flink.test", TestIcebergCommitter.class.getName()),
+        false,
+        10,
+        "sinkId",
+        metric);
+  }
+
+  private Committer.CommitRequest<IcebergCommittable> buildCommitRequestFor(
+      String myJobID, long checkpoint, Collection<WriteResult> writeResults) throws IOException {
+    IcebergCommittable commit =
+        new IcebergCommittable(
+            buildIcebergWriteAggregator(myJobID, "flink-sink")
+                .writeToManifest(writeResults, checkpoint),
+            myJobID,
+            "flink-sink",
+            checkpoint);
+    CommittableWithLineage committableWithLineage =
+        new CommittableWithLineage(commit, checkpoint, 1);
+    Committer.CommitRequest<IcebergCommittable> commitRequest = mock(Committer.CommitRequest.class);
+    doReturn(committableWithLineage.getCommittable()).when(commitRequest).getCommittable();
+    return commitRequest;
+  }
+
+  private WriteResult of(DataFile dataFile) {
+    return WriteResult.builder().addDataFiles(dataFile).build();
+  }
+
+  private void assertMaxCommittedCheckpointId(String myJobID, String operatorId, long expectedId) {
+    table.refresh();
+    long actualId =
+        IcebergCommitter.getMaxCommittedCheckpointId(table, myJobID, operatorId, branch);
+    assertThat(actualId).isEqualTo(expectedId);
+  }
+
+  private void assertMaxCommittedCheckpointId(String myJobID, long expectedId) {
+    assertMaxCommittedCheckpointId(myJobID, "flink-sink", expectedId);
+  }
+
+  private void assertSnapshotSize(int expectedSnapshotSize) {
+    table.refresh();
+    assertThat(table.snapshots()).hasSize(expectedSnapshotSize);
+  }
+
+  private static ByteBuffer longToBuffer(long value) {
+    return ByteBuffer.allocate(8).order(ByteOrder.LITTLE_ENDIAN).putLong(0, value);
+  }
+
+  final TypeSerializer<CommittableMessage<IcebergCommittable>> committableMessageTypeSerializer =
```

Comment:
looks like this is a class member. for readability, maybe define a private static class here and move the class variable up near the beginning, which is typically class member variables are defined.